src/prefect/agent.py

```diff
@@ -194,42 +194,44 @@ async def get_and_submit_flow_runs(self) -> List[FlowRun]:
 
         submittable_runs: List[FlowRun] = []
 
-        if self.work_pool_name and not self.work_queues:
+        if self.work_pool_name:
             responses = await self.client.get_scheduled_flow_runs_for_work_pool_queues(
                 work_pool_name=self.work_pool_name,
+                work_pool_queue_names=self.work_queues,
```

Comment:
We need to call `get_work_queues` to create any work pool queues that don't yet exist and find matching work pool queues if `work_queue_prefix` is used.
```suggestion
                work_pool_queue_names=[wq.name async for wq in self.get_work_queues()],
```

Reply:
ah that's a much better implementation than what I just cooked up; will reset to yours in 1 sec